src/dbnode/storage/index/block.go

```diff
@@ -128,32 +215,254 @@ func (b *block) EndTime() time.Time {
 	return b.endTime
 }
 
-func (b *block) WriteBatch(inserts *WriteBatch) (WriteBatchResult, error) {
+func (b *block) maybeBackgroundCompactWithLock() {
+	if b.compactingBackground || b.state != blockStateOpen {
+		return
+	}
+
+	// Create a logical plan
+	segs := make([]compaction.Segment, 0, len(b.backgroundSegments))
+	for _, seg := range b.backgroundSegments {
+		segs = append(segs, compaction.Segment{
+			Age:     seg.Age(),
+			Size:    seg.Segment().Size(),
+			Type:    segments.FSTType,
+			Segment: seg.Segment(),
+		})
+	}
+
+	plan, err := compaction.NewPlan(segs, b.opts.BackgroundCompactionPlannerOptions())
```

Comment:
Do you need to check if the segments are `Compactable` first? I I saw that method in `plan.go` but it doesn't seem to be used anywhere

Reply:
The compaction planner is the one that actually looks at that and then will return it as "UnusedSegments" if they aren't actually compactable.

Reply:
Are you certain? I did a quick grep of the codebase and can't find any code that calls `segment.Compactable()` except for `plan_test.go`

Reply:
I'll just remove `Compactable`, it doesn't really make sense for us.

Reply:
Done.